File: javatest/src/main/java/com/anymore/javatest/di/module/AppModule.java
package com.anymore.javatest.di.module;

import android.app.Application;
import dagger.Module;
import dagger.Provides;

/**
 * Created by liuyuanmao on 2019/2/22.
 */
@Module
public class AppModule {
    private Application application;

    public AppModule(Application application) {
        this.application = application;
    }

    @Provides
    public Application provideApplication(){
        return application;
    }
}

File: app/src/main/java/com/anymore/simple/di/module/AppModule.kt
package com.anymore.simple.di.module

import android.app.Application
import com.anymore.mvvmkit.di.module.ViewModelFactoryModule
import dagger.Module
import dagger.Provides

/**
 * Created by liuyuanmao on 2019/2/21.
 */
@Module(includes = [ViewModelFactoryModule::class,MainViewModelModule::class])
class AppModule(private val application: Application) {
    @Provides
    fun provideApplication()=application
}